basic-search-on-docker-swarm/docker-compose.yml

```diff
@@ -57,15 +54,68 @@ services:
       - data:/opt/vespa/var
       - logs:/opt/vespa/logs
 
-  content:
+  container1:
     image: vespaengine/vespa
     command: services
-    hostname: "{{.Service.Name}}.{{.Task.Slot}}.{{.Task.ID}}.vespa_net"
+    hostname: "container1.vespa_net"
     environment:
-      - VESPA_CONFIGSERVERS=cfg1.vespa_net,cfg2.vespa_net,cfg3.vespa_net
-    deploy:
-       mode: replicated
-       replicas: 3
+      - VESPA_CONFIGSERVERS=config0.vespa_net,config1.vespa_net,config2.vespa_net
+    ports:
+      - target: 8080
+        published: 8081
```

Comment:
I think it is unfortunate that we can not get the load balancer to work so that port 8080 on any of the swarm hosts forwards to a container on 8080. We should probably dig into this to see if it can be resolved, but the proposed solution works.

Reply:
Agreed. However, I personally won't look into the swarm IPVS possibilities, as I want to stick the containers to specific servers in my cluster (via deployment constraints).
I'd like to try something with a software L7 LB; maybe Traefik. If I have a nice solution I can propose a patch.